### PR TITLE
AzureDiskEncryption: Support disk encryption on RHEL 7.4

### PR DESCRIPTION
**Pull request by mayank88mahajan**

### CODE REVIEW

File: VMEncryption/main/oscrypto/rhel_72/encryptstates/UnmountOldrootState.py [hunk 1]
```diff
@@ -30,6 +30,7 @@ class UnmountOldrootState(OSEncryptionState):
     def __init__(self, context):
         super(UnmountOldrootState, self).__init__('UnmountOldrootState', context)
 
+
```
Comment (vermashi) [inline VMEncryption/main/oscrypto/rhel_72/encryptstates/UnmountOldrootState.py]: Nitpick: Extra blank line.
Source: https://www.python.org/dev/peps/pep-0008/#blank-lines

File: VMEncryption/main/oscrypto/rhel_72/encryptstates/UnmountOldrootState.py [hunk 2]
```diff
@@ -112,12 +118,37 @@ def enter(self):
...
+        proc_comm = ProcessCommunicator()
+        self.command_executor.ExecuteInBash(
+                command_to_execute="grep {0} /proc/*/task/*/mountinfo".format(self.rootfs_block_device),
+                raise_exception_on_failure=True,
+                communicator=proc_comm)
+
+        procs_to_kill = filter(lambda path: path.startswith('/proc/'), proc_comm.stdout.split())
+        procs_to_kill = map(lambda path: int(path.split('/')[2]), procs_to_kill)
+        procs_to_kill = reversed(sorted(procs_to_kill))
+        self.context.logger.log("Processes with tasks using {0}:\n{1}".format(self.rootfs_block_device, procs_to_kill))
```
Comment (vermashi) [inline VMEncryption/main/oscrypto/rhel_72/encryptstates/UnmountOldrootState.py]: How well does this logging statement work? In python 3, the reversed and sorted functions will return lazy objects. Printing them will result in something like: <list_reverseiterator object at 0x000001F987BAA8D0>

Comment (vermashi) [reply VMEncryption/main/oscrypto/rhel_72/encryptstates/UnmountOldrootState.py]: Also, this whole chunk took a while to understand. Some comments explaining how the "procs_to_kill" are selected would be helpful here.

Comment (paulmey) [reply VMEncryption/main/oscrypto/rhel_72/encryptstates/UnmountOldrootState.py]: Maybe add a list() around the last reversed() to force execution then...

File: VMEncryption/main/Common.py [hunk 3]
```diff
@@ -20,7 +20,7 @@
 class CommonVariables:
     utils_path_name = 'Utils'
     extension_name = 'AzureDiskEncryptionForLinuxTest'
```
Comment (vermashi) [inline VMEncryption/main/Common.py]: Remove the Test at the end of the extension_name

Comment (mayank88mahajan) [reply VMEncryption/main/Common.py]: This is done for test publish till we fix our EDP pipeline.

**Pull request merged**